go/timer/randticker.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package timer
+
+import (
+	"math/rand"
+	"time"
+)
+
+var rnd = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+// RandTicker is just like time.Ticker, except that
+// it adds randomness to the events.
+type RandTicker struct {
+	C    <-chan time.Time
+	done chan struct{}
+}
+
+// NewRandTicker creates a new RandTicker. d is the duration,
+// and variance specifies the variance. The ticker will tick
+// every d +/- variance.
+func NewRandTicker(d, variance time.Duration) *RandTicker {
+	c := make(chan time.Time, 1)
+	done := make(chan struct{})
+	go func() {
+		for {
+			vr := time.Duration(rnd.Int63n(int64(2*variance)) - int64(variance))
+			tmr := time.NewTimer(d + vr)
```

Comment:
Caution: I am not quite familiar with select & timer. One question, here why we want to create a new timer for each for loop ? I feel both vr and tmr need to be defined only once, outside the for loop. Again, I am a newbie.